src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -2214,12 +2232,16 @@ void RadioSpinel<InterfaceType, ProcessContextType>::RecoverFromRcpFailure(void)
     mIsReady      = false;
     mIsTimeSynced = false;
 
+#if OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
```

Comment:
In the case of failures on multi-PAN RCP, I don't think we can recover transparently. `OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE` and `OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT > 0` should be mutually exclusive.

Reply:
I think it's necessary to enable the RCP restoration mechanism because the RCP may be reset by another host process.  The key point is that upon receipt of an unexpected reset, the host should not reset the RCP again during recovery.  I see that the current change I made here is too general because it prevents reset in the case of an RCP timeout as well as an unexpected reset.  I can fix this by changing mRcpFailed from a boolean to an enum. How does that sound?

Reply:
The host applications need to coordinate with each about long-term global properties of the radio such as channel and power and enable/disable.  In addition, a policy system can be established in the platform layer on the RCP, for example, only one of the processes has permission to change these.  In any case, it is out of scope of the core stack.

For per-IID properties such as pan id, long id, short id, and the src match lists, storing these separately for each IID is also accomplished at the platform layer.  When handling a set or get command, the PAL looks up the current IID via otNcpPlatGetCurCommandIid().

Reply:
Will the property handling be implemented in this PR or in a separate PR? Looks like the multi PAN feature cannot work properly without coordination between hosts. I think it's good to add some integration test cases so we can better understand the use cases.

Reply:
Will the property handling be implemented in this PR or in a separate PR? Looks like the multi PAN feature cannot work properly without coordination between hosts. I think it's good to add some integration test cases so we can better understand the use cases.

Reply:
The property handling is in the platform layer so it will be in a separate PR.  Testing requires the platform changes and a mux to combine the spinel host streams.  We perform this testing, but I will also look into how this could be added to the openthread testing.  Can you point me to some existing test examples for reference?

Reply:
You can see <https://github.com/openthread/openthread/tree/main/tests/scripts/thread-cert> for virtual time tests and see <https://github.com/openthread/openthread/tree/main/tests/scripts/expect> for real time tests, specifically <https://github.com/openthread/openthread/blob/main/tests/scripts/expect/posix-rcp-restoration.exp>, which contains manual control of RCP and host processes.